pkg/spoke/controllers/finalizercontroller/appliedmanifestwork_finalize_controller.go

```diff
@@ -96,12 +105,25 @@ func (m *AppliedManifestWorkFinalizeController) syncAppliedManifestWork(ctx cont
 
 	owner := helper.NewAppliedManifestWorkOwner(appliedManifestWork)
 
+	var executor *workapiv1.ManifestWorkExecutor
+	manifestWork, err := m.manifestWorkLister.Get(appliedManifestWork.Spec.ManifestWorkName)
+	switch {
+	case err == nil:
+		executor = manifestWork.Spec.Executor
+	case !errors.IsNotFound(err):
+		// if not found the manifest work, keep the executor nil, we will not
+		// check delete permission when executor is nil
+		return err
+	}
+
 	// Work is deleting, we remove its related resources on spoke cluster
 	// We still need to run delete for every resource even with ownerref on it, since ownerref does not handle cluster
 	// scoped resource correctly.
 	reason := fmt.Sprintf("manifestwork %s is terminating", appliedManifestWork.Spec.ManifestWorkName)
-	resourcesPendingFinalization, errs := helper.DeleteAppliedResources(
-		ctx, appliedManifestWork.Status.AppliedResources, reason, m.spokeDynamicClient, controllerContext.Recorder(), *owner)
+	resourcesPendingFinalization, errs := spokehelper.DeleteAppliedResources(
```

Comment:
we also need to update condition for work here, to declare the remaining resource and err